src/cdk-experimental/tree/navigator.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {LEFT_ARROW, RIGHT_ARROW, UP_ARROW, DOWN_ARROW, HOME, END} from '@angular/cdk/keycodes';
+import {Directive} from '@angular/core';
+import {CdkTreeNode} from './node';
+import {CdkTree} from './tree';
+
+
+/**
+ * CDK tree component that connects with a data source to retrieve data of type `T` and renders
+ * dataNodes with hierarchy. Updates the dataNodes when new data is provided by the data source.
+ */
+@Directive({
+  selector: '[cdkTreeNavigator]',
+  host: {
+    '(keydown)': '_handleKeydown($event)'
+  }
+})
+export class CdkTreeNavigator<T> {
+  focusedData: T;
+
+  constructor(protected _tree: CdkTree<T>) {}
+
+  /** Change focus to first visible node in the tree. */
+  focusFirst() {
+    this._changeFocusedData(this._tree.treeControl.getFirstNode());
+  }
+
+  /** Change focus to last visible node in the tree. */
+  focusLast() {
+    this._changeFocusedData(this._tree.treeControl.getLastNode());
+  }
+
+  /** Change focus to previous visible node. */
+  focusPreviousVisibleNode() {
```

Comment:
Maybe just` focusPreviousNode` where the "visible" is implicit (and mentioned in the description)?

(same for focusNextNode)